libs/permissions/permission-storage-writer-impl/src/main/kotlin/net/corda/libs/permissions/storage/writer/impl/validation/EntityValidationUtil.kt

```diff
@@ -74,6 +74,28 @@ class EntityValidationUtil(private val entityManager: EntityManager) {
         }
     }
 
+    fun validateNewParentGroupNotADescendant(groupToModify: String, newParentGroupId: String) {
+        val query = "SELECT g.id, g.parent_group FROM rbac_group g"
+        val groupMap = entityManager.createNativeQuery(query)
```

Comment:
Shall we avoid native queries? e.g.:
```suggestion
        val groupMap = entityManager.createQuery("FROM Group", Group::class.java)
```

Reply:
Actually, will it not be more efficient to implement this sort of check at RestWorker side using `net.corda.libs.permissions.management.cache.PermissionManagementCache#groups`?
It will not require an expensive `select` into memory with every call and Map of groups should be fairly up-to-date in the cache to perform this sort of a check.

Reply:
Ah ok, I had thought about this initially, but I wasn't confident in depending on the cache for this. I was also thinking to keep the validation for these operations on the DB side with our other validation. But neither of these are very strong points. I'll change to use the cache if we're happy to do it that way. Thanks for the review.